Library/Homebrew/formula_creator.rb

```diff
@@ -57,6 +50,7 @@ def head?
     end
 
     def generate!
+      path = @tap.new_formula_path(@name)
```

Comment:
What happens if `@tap` or `@name` are `nil` here?

Reply:
If @tap is `nil`, the code will fail with something like `nil` object has no method `new_formula_path`. In the previous version if `@tap` or `@name` is `nil`, then `path` would be `nil` and fail on the next line where `path.exists?` is called.

Reply:
In the previous version `update_path` was at least checking for these being `nil`. I think it would be good to handle that here, too.

Reply:
The check didn't do anything useful. `@path` remained `nil` and then would fail at the next line.

That would fail only if having `@tap` or `@name` `nil` at this point is at all possible. `create.rb` contains checks at least for `@name` that probably won't pass it to be empty. This need to be tested.

Reply:
Yes, I believe the pathway where `@tap` is `nil` is unreachable.

```
$ brew create --set-name= --set-version=1 https://github.com/lapce/lapce\?name\= --tap=
Formula name []: 
Error: Invalid tap name ''
```

Reply:
An empty `@name` is however possible at `master`.

```
$ brew create --set-name= --set-version=1 https://github.com/lapce/lapce\?name\=        
Formula name []: 
==> Downloading https://github.com/lapce/lapce?name=
Already downloaded: /home/anatoli/.cache/Homebrew/downloads/392c0a11eadf32d86b1c4c5dc48d6443e52af912d3bc95b73dac6a511fe0351d--lapce
Warning: Cannot verify integrity of '392c0a11eadf32d86b1c4c5dc48d6443e52af912d3bc95b73dac6a511fe0351d--lapce'.
No checksum was provided.
For your reference, the checksum is:
  sha256 "d26f8dfbd3bb6ace122c5a032c6cedcbf1cf3921f19493ca652c4270261df6fe"
Error: : /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/.rb:4: syntax error, unexpected '<'
class  < Formula
       ^
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/.rb:34: syntax error, unexpected end, expecting end-of-input
```

Reply:
Would still be good to add checks here, please.

Reply:
Added the check for an empty name. 

But there is still no check for invalid names, like "34" that will result in writing invalid ruby file. If there is a name validator function, it would be better to call it here.